<launch>
  
  <!-- LAUNCH INTERFACE FOR FT SENSORS -->

  <arg name="ip" default="172.16.0.120" doc="Network address of the sensor. Mandatory"/>
  <arg name="name" default="/setup1/sensor2" doc="Name of the sensor for unique identification, it depends on your URDF implementation. Mandatory."/>
  <arg name="type" default="nano17" doc="Type of sensor, e.g. nano17, it depends on your URDF implementation, check which models this package supports. Mandatory."/>
  <arg name="frame_id" default="right_arm_7_link" doc="Reference frame for the sensor"/>
  <arg name="safety_topic" default="/right_arm/emergency_event" doc="Topic to advertise for a emergency event"/>
  <arg name="safety_threshold" default="0.7" doc="max normalized wrench allowed before a emergency event"/>
  <arg name="publish_rate" default="500" doc="publish rate for force readings"/>
  <arg name="startDataStream" default="true" doc="start data stream on init. It is used true to set bias on init"/>

  <!-- LAUNCH IMPLEMENTATION -->
  <group ns="$(arg name)">
    <node  name="ft_sensor_hw" pkg="force_torque_sensor" type="ati_sensor_node" respawn="true" output="screen">
      <param name="ip" value="$(arg ip)"/>
      <param name="name" value="ati_FT" />
      <param name="type" value="$(arg type)" />
      <param name="frame_id" value="$(arg frame_id)" />
      <param name="safety_topic" value="$(arg safety_topic)" />
      <param name="safety_threshold" value="$(arg safety_threshold)" />
      <param name="publish_rate" value="$(arg publish_rate)" />
      <param name="startDataStream" value="$(arg startDataStream)" />
    </node>
    <node  name="ati_bridge" pkg="force_torque_sensor" type="ati_bridge_node" respawn="true" output="screen"/>
  </group>
   
</launch>